### PR TITLE
Added TS types to eris-pagination

### PR DESCRIPTION
**Pull request by CircuitRCAY**
This will obviously bring easier TypeScript support to the library.

### CODE REVIEW

**Comment by knht** [comment]
Hey, thanks for the PR! I'll have it reviewed ASAP and have it merged if nothing is wrong with the typings.

File: package.json [hunk 1]
```diff
@@ -26,5 +26,8 @@
   "homepage": "https://github.com/riyacchi/eris-pagination#readme",
   "dependencies": {
     "eris-reactions": "^0.1.2"
+  },
+  "devDependencies": {
+    "eris": "^0.10.1"
   }
```
Comment (eritbh) [inline package.json:32]: Not strictly necessary, but Typescript docs say it's advisable to add the `"types"` key even when the typing file is `index.d.ts`.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+declare module 'eris-pagination' {
+    import { Message, EmbedOptions } from 'eris';
+
+    class PaginationEmbed {
+        constructor(message: Message, pages: EmbedOptions[], options?: PaginationEmbedOptions);
+        public initialize(): Promise<void>;
+        public run(): void;
+        public update(): void;
+    }
```
Comment (eritbh) [inline index.d.ts:14]: This class definition isn't necessary since it isn't accessible to API consumers. It should be removed.

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        timeout?: number;
+        deleteButton?: string;
+        firstButton?: string;
+        lastButton?: string;
+        backButton?: string;
+        forthButton?: string;
+        startPage?: number;
+    }
+
+    export function createPaginationEmbed(message: Message, pages: EmbedOptions[], options?: PaginationEmbedOptions): Promise<void>;
```
Comment (eritbh) [inline index.d.ts:29]: `export` is implied inside `declare` blocks. Definitions should be consistent about using it - either include or omit the `export` keyword on everything inside `declare`. I suggest using it on everything for clarity.

**Review by eritbh — CHANGES_REQUESTED** [review]

**Comment by bsian03** [comment]
The user looks like they deleted their fork of the repository, I just wrote up my own typings file, do you want me to submit a PR?

**Comment by knht** [comment]
> The user looks like they deleted their fork of the repository, I just wrote up my own typings file, do you want me to submit a PR?

That'd be really appreciated!

**Comment by knht** [comment]
Closing due to inactivity.

**Pull request closed without merging**